packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.tsx

```diff
@@ -54,33 +54,40 @@ export default defineComponent({
 
     const isActive = useProxiedModel(props, 'modelValue')
     const isHovering = ref(false)
-    const isMobile = display.value.mobile
     const size = computed(() => Number(props.rail ? props.railWidth : props.width))
     const width = computed(() => {
       return (props.rail && props.expandOnHover && isHovering.value)
         ? props.width
         : size.value
     })
+    const isMobile = computed(() => display.value.mobile)
+    const isTemporary = computed(() => !props.permanent && (isMobile.value || props.temporary))
     const layoutStyles = useLayoutItem(
       props.name,
       toRef(props, 'priority'),
       computed(() => props.right ? 'right' : 'left'),
-      computed(() => {
-        return (
-          props.permanent ||
-          (isActive.value && !props.temporary)
-        ) ? size.value : 0
-      }),
+      computed(() => !isTemporary.value ? size.value : 0),
     )
 
+    if (!props.disableResizeWatcher) {
+      watch(isMobile, val => !props.permanent && (isActive.value = !val))
+    }
+
+    watch(props, val => {
```

Comment:
1. This is why I think `permanent` is a pointless prop. You could just do `:model-value="true"`

(If you really don't want to drop it, the it should probably be `watch(() => props.permanent, ...)`

Reply:
The **permanent** prop is more for semantic purposes and I would like to keep it. I'll update the code to match your suggestion.